python-sdk/nuscenes/eval/lidarseg/render.py

```diff
@@ -0,0 +1,327 @@
+        """
+        for i, strata in enumerate(self.strata_list):
+            print('Evaluating for strata {}...'.format(self.strata_names[i]))
+            for sample_token in tqdm(self.sample_tokens, disable=not self.verbose):
+                sample = self.nusc.get('sample', sample_token)
+
+                # 1. Get the sample data token of the point cloud.
+                sd_token = sample['data']['LIDAR_TOP']
+                pointsensor = self.nusc.get('sample_data', sd_token)
+                pcl_path = os.path.join(self.nusc.dataroot, pointsensor['filename'])
+
+                # 2. Load the ground truth labels for the point cloud.
+                gt_path = os.path.join(self.nusc.dataroot, self.nusc.get('lidarseg', sd_token)['filename'])
+                gt = LidarSegPointCloud(pcl_path, gt_path)
+                gt.labels = self.mapper.convert_label(gt.labels)  # Map the labels as necessary.
+
+                # 3. Load the predictions for the point cloud.
+                pred_path = os.path.join(self.results_folder, 'lidarseg', self.eval_set, sd_token + '_lidarseg.bin')
+                pred = LidarSegPointCloud(pcl_path, pred_path)
+
+                # 4. Filter to get only labels belonging to the strata.
+                gt = self.filter_pointcloud_by_depth(gt, min_depth=strata[0], max_depth=strata[1])
+                pred = self.filter_pointcloud_by_depth(pred, min_depth=strata[0], max_depth=strata[1])
+
+                # 5. Update the confusion matrix for the sample data into the confusion matrix for the eval set.
+                self.global_cm[i].update(gt.labels, pred.labels)
+
+        self.stratified_per_class_metrics = self.get_stratified_per_class_metrics()
+        if self.verbose:
+            print(json.dumps(self.stratified_per_class_metrics, indent=4))
+
+        if self.output_dir:
+            self.render_stratified_per_class_metrics(os.path.join(self.output_dir, 'stratified_iou_per_class.png'))
+            self.render_stratified_overall_metrics(os.path.join(self.output_dir, 'stratified_iou_overall.png'))
+
+    def render_stratified_overall_metrics(self, filename: str, dpi: int = 100) -> None:
+        """
+        Renders the stratified overall metrics (i.e. the classes are aggregated for each strata).
+        :param filename: Filename to save the render as.
+        :param dpi: Resolution of the output figure.
+        """
+        stratified_iou = {strata_name: self.stratified_per_class_metrics[i]['miou'] for i, strata_name in
+                          enumerate(self.strata_names)}
+
+        fig = plt.figure()
+        ax = fig.add_subplot(111)
+        ax.bar(list(stratified_iou.keys()),
+               list(stratified_iou.values()), color='grey')
+        ax.set_xlabel('Interval / m', fontsize=15)
+        ax.set_ylabel('mIOU', fontsize=15)
+        ax.set_ylim(top=1.1)  # Make y-axis slightly higher to accommodate tag.
+        ax.set_title('Stratified results for mIOU', fontsize=20)
```

Comment:
"stratified results" is a bit vague.
`Distance vs. <metric> for <class>`
<class>: class name or "all classes"
<metric>: IOU or mIOU

Reply:
Amended title as per your suggestion